ui/redux/rootReducer.js

```diff
@@ -187,85 +180,22 @@ export const updateGeneNote = values => updateEntity(
   values, RECEIVE_DATA, `/api/gene_info/${values.geneId || values.gene_id}/note`, 'noteGuid',
 )
 
-export const navigateSavedHashedSearch = (search, navigateSearch, resultsPath, hashKey) => (dispatch) => {
+const navigateSavedHashedSearchHelper = (resultsPath, hashKey) => (search, navigateSearch) => (dispatch) => {
```

Comment:
this is actually a small code change- the way this function was defined allowed any caller to mutate an arbitrary property of the reducer state which makes it nearly impossible to tell which pieces of the state are accessed and modified where. This instead makes a new helper that explicitly modifies that part of the state that was being passed as `hashKey` before